### DISCUSSION TITLE
Roku Cam

### DISCUSSION THREAD

**Discussion by skilau** [gtxaspec/wz_mini_hacks#379] [General]
As most know, Roku recently took Wyze cams/lights/etc, and slapped their name on them, and are selling them in places like Walmart now:
https://www.walmart.com/ip/Roku-Smart-Home-Indoor-Camera-SE-Wired-Security-Camera-with-Motion-Sound-Detection/850535193

The price ends up a bit cheaper by 3-4 bucks compared to regular Wyze v3 at my local Microcenter.

The question is, do they also have the flaw/bug in their bootloader that lets us hijack the boot process and install our own stuff?

Might be an interesting experiment...

**Comment by gtxaspec** [comment]
don't have a device, need someone to capture the firmware URL or dump the rom chip from the camera.

if someone has it, put a microsd card with wz_mini installed and see if it boots.

**Reply by KevinKadow** [reply]
> don't have a device, need someone to capture the firmware URL or dump the rom chip from the camera.

Update server is prod-firmware-upgrade-service.wyzeiot.com for firmware.  I wasn't able to sniff the full firmware URL on the network because the camera will just abort the connection if it doesn't see a trusted server certificate after the initial SSL handshake.

After the mandatory first-use update  reports firmware version "3.01.2632". 

> if someone has it, put a microsd card with wz_mini installed and see if it boots.

In my testing, Roku camera doesn't recognize the mini-hacks files.  

I pulled the card and it created all the usual folders (including .ota) as well as a folder named "rebootlog" containing an empty file.  Also placing "demo_wcv3.bin" in the root folder and restarting with setup button held down does nothing.  I suspect it looks for an entirely different file and filename.

According to the Wyze Discord, the devices are not firmware-compatible, the Wyze firmware will not run on Ruko-branded hardware nor vice-versa.

**Reply by mach7** [reply]
@gtxaspec how would we dump a rom chip? I'd be willing to go purchase one to play with it. I'm a stay at home dad with a little free time on my hands to help.

**Comment by KevinKadow** [comment]
> if someone has it, put a microsd card with wz_mini installed and see if it boots.

My local Walmart has them in stock for immediate pickup, I may give it a go.

Anything specific I should do at first boot or to "_capture the firmware URL_"?    Looking at the Roku support site, appears they have yet to release any updates to the stock firmware.

**Comment by akivacp** [comment]
Would like to see if Roku cameras are, indeed, supported. Walmart has 2-pack on sale right now for the price of 1 Wyze cam. The hack does not work.
https://www.walmart.com/ip/Roku-Smart-Home-Indoor-Camera-SE-2-Pack-Wi-Fi-Connected-Wired-Security-Surveillance-Camera-with-Motion-Sound-Detection/559118776

**Reply by KevinKadow** [reply]
So far no joy.    

The firmware is specific to Roku, camera doesn't react at all to a MicroSD card with wz_mini_hacks loaded on the card.

**Comment by teredactle** [comment]
@KevinKadow does the ROKU version have firmware with RTSP support by any chance?? This would be amazing!

**Comment by DAMAGEDatheCORE** [comment]
I doesn't, no.
